test/unit/template_importer_test.rb

```diff
@@ -0,0 +1,96 @@
+require 'test_plugin_helper'
+
+module ForemanTemplates
+  class TemplateImporterTest < ActiveSupport::TestCase
+    def get_template(name)
+      File.expand_path(File.join('..', '..', 'templates', name), __FILE__)
+    end
+
+    def importer(opts = {})
+      # This uses a checkout of the plugin as a source of templates Caveat:
+      # changes to /test/templates will need to be committed (locally) for
+      # tests to work
+
+      path = File.expand_path(File.join('..','..','..'),__FILE__)
+      ForemanTemplates::TemplateImporter.new({
+        :repo      => path,
+        :branch    => Git.open(path).current_branch,
+        :prefix    => 'FooBar ',
+        :dirname   => '/test/templates',
+        :verbose   => 'false',
+        :associate => 'new'
+      }.merge(opts))
+    end
+
+    setup do
+      @importer = importer
+    end
+
+    context 'get_default_branch' do
+      setup do
+        @repo = Struct.new(:branches).new([
+          Struct.new(:name).new('0.1-stable'),
+          Struct.new(:name).new('develop')
+        ])
+      end
+
+      test 'when on develop, returns develop' do
+        SETTINGS[:version].stubs(:tag).returns('develop')
+        assert_equal 'develop', @importer.get_default_branch(@repo)
+      end
+
+      test 'when branch exists, return it' do
+        SETTINGS[:version].stubs(:tag).returns('not_develop')
+        SETTINGS[:version].stubs(:short).returns('0.1')
+        assert_equal '0.1-stable', @importer.get_default_branch(@repo)
+      end
+
+      test 'when branch does not exist, use default branch' do
+        SETTINGS[:version].stubs(:tag).returns('not_develop')
+        SETTINGS[:version].stubs(:short).returns('0.2')
+        refute @importer.get_default_branch(@repo)
+      end
+    end
+
+    test 'metadata method extracts correct metadata' do
+      text = File.read(get_template('metadata1.erb'))
+      hash = @importer.parse_metadata(text)
+      assert_equal 'provision', hash['kind']
+      assert_equal 'Test Data', hash['name']
+      assert_equal 5, hash['oses'].size
+
+      text = File.read(get_template('metadata2.erb'))
+      hash = @importer.parse_metadata(text)
+      assert_equal 'ProvisioningTemplate', hash['model']
+    end
+
+    test 'import! loads templates into db' do
```

Comment:
Sure first rebase was just to shoe horn purge in.

Thanks for the pointers, I'll try and squeeze it in tomorrow.

On 31 August 2016 at 20:00, Greg Sutcliffe <notifications@github.com> wrote:

> In test/unit/template_importer_test.rb
> <https://github.com/theforeman/foreman_templates/pull/29#discussion_r77051307>
> :
>
> > +      end
> > +    end
> > +
> > +    test 'metadata method extracts correct metadata' do
> > +      text = File.read(get_template('metadata1.erb'))
> > +      hash = @importer.parse_metadata(text)
> > +      assert_equal 'provision', hash['kind']
> > +      assert_equal 'Test Data', hash['name']
> > +      assert_equal 5, hash['oses'].size
> > +
> > +      text = File.read(get_template('metadata2.erb'))
> > +      hash = @importer.parse_metadata(text)
> > +      assert_equal 'ProvisioningTemplate', hash['model']
> > +    end
> > +
> > +    test 'import! loads templates into db' do
>
> Again, if you need a hand with that, let me know.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/theforeman/foreman_templates/pull/29/files/896c6f18905e9e84edcd9ca2eed82dedb70761ad#r77051307>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABx8sgP1XLj8xgDWqM_d7Cs4VO7DtVJ6ks5qlc9CgaJpZM4JxgHW>
> .
>